fabrics.c

```diff
@@ -86,31 +86,58 @@ static const char *nvmf_data_digest	= "enable transport protocol data digest (TC
-	OPT_FLAG("tls",                 0, &c.tls,                nvmf_tls)		\
+#define NVMF_OPTS(args) {                                                                       \
+	OPT_STRING("transport",       't', "STR", args->transport,       nvmf_tport),           \
+	OPT_STRING("nqn",             'n', "STR", args->subsysnqn,       nvmf_nqn),             \
+	OPT_STRING("traddr",          'a', "STR", args->traddr,          nvmf_traddr),          \
+	OPT_STRING("trsvcid",         's', "STR", args->trsvcid,         nvmf_trsvcid),         \
+	OPT_STRING("host-traddr",     'w', "STR", &args->c->host_traddr, nvmf_htraddr),         \
+	OPT_STRING("host-iface",      'f', "STR", &args->c->host_iface,  nvmf_hiface),          \
+	OPT_STRING("hostnqn",         'q', "STR", args->hostnqn,         nvmf_hostnqn),         \
+	OPT_STRING("hostid",          'I', "STR", args->hostid,          nvmf_hostid),          \
+	OPT_STRING("dhchap-secret",   'S', "STR", args->hostkey,         nvmf_hostkey),         \
+	OPT_INT("nr-io-queues",       'i', &args->c->nr_io_queues,       nvmf_nr_io_queues),    \
+	OPT_INT("nr-write-queues",    'W', &args->c->nr_write_queues,    nvmf_nr_write_queues), \
+	OPT_INT("nr-poll-queues",     'P', &args->c->nr_poll_queues,     nvmf_nr_poll_queues),  \
+	OPT_INT("queue-size",         'Q', &args->c->queue_size,         nvmf_queue_size),      \
+	OPT_INT("keep-alive-tmo",     'k', &args->c->keep_alive_tmo,     nvmf_keep_alive_tmo),  \
+	OPT_INT("reconnect-delay",    'c', &args->c->reconnect_delay,    nvmf_reconnect_delay), \
+	OPT_INT("ctrl-loss-tmo",      'l', &args->c->ctrl_loss_tmo,      nvmf_ctrl_loss_tmo),   \
+	OPT_INT("tos",                'T', &args->c->tos,                nvmf_tos),             \
+	OPT_INT("keyring",              0, &args->c->keyring,            nvmf_keyring),         \
+	OPT_INT("tls_key",              0, &args->c->tls_key,            nvmf_tls_key),         \
+	OPT_FLAG("duplicate-connect", 'D', &args->c->duplicate_connect,  nvmf_dup_connect),     \
+	OPT_FLAG("disable-sqflow",    'd', &args->c->disable_sqflow,     nvmf_disable_sqflow),  \
+	OPT_FLAG("hdr-digest",        'g', &args->c->hdr_digest,         nvmf_hdr_digest),      \
+	OPT_FLAG("data-digest",       'G', &args->c->data_digest,        nvmf_data_digest),     \
+	OPT_FLAG("tls",                 0, &args->c->tls,                nvmf_tls),             \
+	OPT_END()                                                                               \
+}
+
+struct nvmf_args {
+	char **transport;
+	char **subsysnqn;
+	char **traddr;
+	char **trsvcid;
+	char **hostnqn;
+	char **hostid;
+	char **hostkey;
+	struct nvme_fabrics_config *c;
+	struct argconfig_commandline_options *opts;
+};
+
+#define NVMF_ARGS(args) \
+	struct nvmf_args args = { \
+		.transport = &transport, \
+		.subsysnqn = &subsysnqn, \
+		.traddr = &traddr, \
+		.trsvcid = &trsvcid, \
+		.hostnqn = &hostnqn, \
+		.hostid = &hostid, \
+		.hostkey = &hostkey, \
+		.c = &cfg, \
+		.opts = opts, \
+	}
```

Comment:
... and this wouldn't be necessary. So there wouldn't be any local open coded variables. Only the `args` struct. Obviously all commands need to be updated where we currently do 

` nvme_foo(transport, subsysysnqn) -> nvme_foo(args.transport, args.subsysnqn)`

Reply:
Hmm, but I don't think this makes actually sense. As we always have some commends with different arguments.

So your patch seems reasonable. Maybe we could also introduce a macro which defines all these common variables like hostnqn etc which are expected by NVMF_ARGS.

Reply:
But given that the goal here is just to get rid of the checkpatch warning, I find it a bit too much code churn. Would the first part of this patch be enough? I mean the change to

`#define NVME_OPTS(args)` ->  `#define NVMF_OPTS(args) {`

Reply:
No only the first part below causes a build error.
```c
#define NVMF_OPTS(args) {                                                                       \
	OPT_STRING("transport",       't', "STR", args->transport,       nvmf_tport),           \
...
	OPT_FLAG("tls",                 0, &args->c->tls,                nvmf_tls),             \
}
...
static int discover_from_conf_file(nvme_root_t r, nvme_host_t h,
				   const char *desc, bool connect,
				   const struct nvme_fabrics_config *defcfg)
{
...
	OPT_ARGS(opts) = {
		NVMF_OPTS(cfg),
		OPT_FMT("output-format", 'o', &format,        output_format),
...
		OPT_FLAG("force",          0, &force,      "Force persistent discovery controller creation"),
		OPT_END()
	};
```
Since the macro is expanded as below but the `{` and `}` expanded are not able to be used in side of the OPT_ARGS `{` and `}`.
```c
static int discover_from_conf_file(nvme_root_t r, nvme_host_t h,
				   const char *desc, bool connect,
				   const struct nvme_fabrics_config *defcfg)
{
...
	OPT_ARGS(opts) = {
		{
		OPT_STRING("transport",       't', "STR", args->transport,       nvmf_tport),
...
		OPT_FLAG("tls",                 0, &args->c->tls,                nvmf_tls),
		}
		OPT_FMT("output-format", 'o', &format,        output_format),
...
		OPT_FLAG("force",          0, &force,      "Force persistent discovery controller creation"),
		OPT_END()
	};

...